cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java

```diff
@@ -306,23 +318,31 @@ public TasksForOffer getTasksForOffer(@NotNull final Protos.Offer offer) {
 
             CassandraNode.Builder node;
             if (!nodeOption.isPresent()) {
+                if (clusterState.get().getNodesToAcquire() <= 0) {
+                    // number of C* cluster nodes already present
+                    return null;
+                }
+
                 NodeCounts nodeCounts = clusterState.nodeCounts();
                 if (nodeCounts.getNodeCount() >= defaultConfigRole.getNumberOfNodes()) {
                     // number of C* cluster nodes already present
                     return null;
                 }
 
+                String replacementForIp = clusterState.nextReplacementIp();
+
                 boolean buildSeedNode = nodeCounts.getSeedCount() < defaultConfigRole.getNumberOfSeeds();
-                CassandraNode newNode = buildCassandraNode(offer, buildSeedNode);
-                clusterState.nodes(append(
-                        clusterState.nodes(),
-                        newNode
-                ));
+                CassandraNode newNode = buildCassandraNode(offer, buildSeedNode, replacementForIp);
+                clusterState.nodeAcquired(newNode);
                 node = CassandraNode.newBuilder(newNode);
             } else
                 node = CassandraNode.newBuilder(nodeOption.get());
 
             if (!node.hasCassandraNodeExecutor()) {
+                if (node.getTargetRunState() == CassandraNode.TargetRunState.TERMINATE) {
```

Comment:
Do terminated nodes stick around in the state info after we've verified that the node has terminated?  It seems like we're checking the terminated state a lot.